pkg/predicate/predicate.go

```diff
@@ -31,11 +27,7 @@ func (OnlyUpdateWhenGenerationNotChanged) Update(e event.UpdateEvent) bool {
 		log.Error(nil, "Update event has no new runtime object for update", "event", e)
 		return false
 	}
-	if e.MetaNew == nil {
-		log.Error(nil, "Update event has no new metadata", "event", e)
-		return false
-	}
-	return e.MetaNew.GetGeneration() == e.MetaOld.GetGeneration()
+	return e.ObjectNew.GetGeneration() == e.ObjectOld.GetGeneration()
```

Comment:
just curious: what was wrong with the 2 cases that you deleted? didn't they apply to creation and deletion?

Reply:
that's the result of the API changes in the latest k8s - the objects that are provided by the event is not a `runtime.Client` but `client.Client`

Reply:
I'm not sure I understand your response. My question was about the removal of
```
if e.MetaOld == nil {
	log.Error(nil, "Update event has no old metadata", "event", e)
	return false
}
...
if e.MetaNew == nil {
	log.Error(nil, "Update event has no new metadata", "event", e)
	return false
}
```
in this func.

Reply:
The ` event.UpdateEvent` doesn't contain the `e.MetaOld` nor `e.MetaNew` now. Instead of `runtime.Client` it provides `client.Client` which has already access to all meta information. So, there is no reason for providing that meta thingy as a separate parameter and thus we just check if the `client.Client` objects are nil or not - that's sufficient. If there is no meta, then there is no `client.Client` object.
This same checks are also used in the built-in predicates

Sorry, for not being clear before.

Reply:
ok, thanks for the clarification